src/System.Private.ServiceModel/tools/test/SelfHostWcfService/TestResources/WebSocketResources.cs

```diff
@@ -47,4 +42,54 @@ protected override Binding GetBinding()
             return binding;
         }
     }
+
+    internal class WebSocketHttpDuplexBinaryStreamed : EndpointResource<WSDuplexService, IWSDuplexService>
+    {
+        protected override string Protocol { get { return BaseAddressResource.Http; } }
+
+        protected override int GetPort(ResourceRequestContext context)
+        {
+            return context.BridgeConfiguration.BridgeWebSocketPort;
+        }
+
+        protected override string Address { get { return "WebSocketHttpDuplexBinaryStreamed"; } }
+
+        protected override Binding GetBinding()
+        {
+            int DefaultMaxReceivedMessageSize = 64 * 1024 * 1024;
+
+            NetHttpBinding binding = new NetHttpBinding();
+            binding.WebSocketSettings.TransportUsage = WebSocketTransportUsage.Always;
+            binding.MaxReceivedMessageSize = DefaultMaxReceivedMessageSize;
+            binding.MaxBufferSize = DefaultMaxReceivedMessageSize;
+            binding.TransferMode = TransferMode.Streamed;
+            binding.MessageEncoding = NetHttpMessageEncoding.Binary;
+            return binding;
+        }
+    }
+
+    internal class WebSocketHttpRequestReplyBinaryStreamed : EndpointResource<WSRequestReplyService, IWSRequestReplyService>
+    {
+        protected override string Protocol { get { return BaseAddressResource.Http; } }
+
+        protected override int GetPort(ResourceRequestContext context)
+        {
+            return context.BridgeConfiguration.BridgeWebSocketPort;
+        }
+
+        protected override string Address { get { return "WebSocketHttpRequestReplyBinaryStreamed"; } }
+
```

Comment:
Suggestion only: I see several Address overrides return a literal string that matches their class name.  It might be more maintainable to return this.GetType().Name() from the base class implementation of Address.  That way, most classes would not need to override it.